### PR TITLE
Add Hive caching and azure storage support documentation

### PR DESCRIPTION
**Pull request by mosabua**
Content already reviewed and tested by @sopel39 and others. This is just a contribution to upstream.

Also related https://github.com/prestosql/presto/pull/4229

### CODE REVIEW

File: presto-docs/src/main/sphinx/connector/hive-azure-storage.rst [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+Hive connector configuration
+----------------------------
+
+ All configuration for the Azure storage driver is stored in the Hadoop
+ ``core-site.xml`` configuration file. The path to the file needs to be
+ configured in the the catalog properties file:
+
+.. code-block:: properties
+
+	hive.config.resources=<path_to_hadoop_core-site.xml>
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/hive-azure-storage.rst:28]: Indent four spaces (this looks like a tab)

File: presto-docs/src/main/sphinx/connector/hive-azure-storage.rst [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    - Description
+  * - ``hive.azure.abfs-storage-account``
+    - Storage account name of Azure Data Lake Storage Gen2
+  * - ``hive.azure.abfs-access-key``
+    - The decrypted access key for the Azure Data Lake Storage Gen2ß
+
+While it is advised to migrate to ADLS Gen2 whenever possible, if you still
+choose to use ADLS Gen1 you need to add the following properties.
+
+.. note:: Credentials for the filesystem can be configured using
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/hive-azure-storage.rst:70]: Format multi-line notes like
```
.. note::

    This is a
    long note.

File: presto-docs/src/main/sphinx/connector/hive-azure-storage.rst [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+URI scheme so that resources (directories and files) may be distinctly
+addressed. You can access both primary and secondary storage accounts linked to
+the cluster with the same URI scheme. Following are example URIs for the
+different systems.
+
+ABFS URI:
+
+.. code-block:: none
+
+  abfs[s]://file_system@account_name.dfs.core.windows.net/<path>/<path>/<file_name>
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/hive-azure-storage.rst:107]: Same comments about indentation

File: presto-docs/src/main/sphinx/connector/hive-azure-storage.rst [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    );
+
+Now you can query the newly mapped table::
+
+    SELECT * FROM orders;
+
+Writing data
+------------
+
+**Prerequisites**
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/hive-azure-storage.rst:157]: Make these second level headers with `^^^`

File: presto-docs/src/main/sphinx/connector/hive-storage-caching.rst [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+.. contents::
+  :local:
+  :backlinks: none
+
+Benefits
+--------
+
+Enabling caching can result in significant benefits:
+
+**Reduced load on object storage**
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/hive-storage-caching.rst:21]: Headers with `^^^`

File: presto-docs/src/main/sphinx/connector/hive-storage-caching.rst [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+:doc:`inspect these and other metrics directly in Presto with the JMX connector
+or in external tools </admin/jmx>`.
+
+The following example query returns the total amount of data fetched by the
+BookKeeper on the coordinator and all the separate nodes on the workers in the
+default async mode:
+
+.. code-block:: sql
+
+  SELECT fs.readfromremote + async_downloaded_mb.count
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/hive-storage-caching.rst:183]: Fix indentation

File: presto-docs/src/main/sphinx/connector/hive-azure-storage.rst [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+
+Presto supports both ADLS Gen1 and Gen2. With ADLS Gen2 now generally available,
+we recommend using ADLS Gen2. Learn more from `the official documentation
+<https://docs.microsoft.com/en-us/azure/data-lake-store/data-lake-store-overview>`_.
+
+.. contents::
+  :local:
+  :backlinks: none
+
+Hive connector configuration
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/hive-azure-storage.rst:19]: Uppercase title

Comment (mosabua) [reply presto-docs/src/main/sphinx/connector/hive-azure-storage.rst:19]: I started adopting the Google style .. hence normal sentence case..

**Comment by electrum** [comment]
Merged, thanks!

File: presto-docs/src/main/sphinx/connector/hive-storage-caching.rst [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+
+A result of the reduced load on the object storage, mentioned earlier, is
+significantly reduced network traffic. Network traffic however is a considerable
+cost factor in an setup, specifically also when hosted in public cloud provider
+systems.
+
+Architecture
+------------
+
+Caching for distributed storage provides a read-through cache. After first
```
Comment (sopel39) [inline presto-docs/src/main/sphinx/connector/hive-storage-caching.rst:63]: Default mode in Rubix is `async`. It means that after a while data will be cached asynchronous. Benefit of that is that reads can be merged together and original query doesn't have to wait for data to be cached first.

`read-through` mode of caching works indeed as described here

File: presto-docs/src/main/sphinx/connector/hive-storage-caching.rst [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+
+    connector.name=hive-hadoop2
+    hive.cache.enabled=true
+    hive.cache.location=/opt/hive-cache
+
+The cache operates on the coordinator and all workers accessing the object
+storage. The used networking ports for the managing BookKeeper and the data
+transfer, by default 8898 and 8899, need to be available.
+
+To use caching on multiple catalogs, you need to configure different caching
```
Comment (sopel39) [inline presto-docs/src/main/sphinx/connector/hive-storage-caching.rst:86]: I would skip `different` word. For ports it's obvious that they need to be different. For directories we could mention that more than single caching directory is preferred for performance reasons.

File: presto-docs/src/main/sphinx/connector/hive-storage-caching.rst [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+      More tips can be found in the :ref:`recommendations
+      <hive-cache-recommendations>`.
+    -
+  * - ``hive.cache.data-transfer-port``
+    -  The TCP/IP port used to transfer data managed by the cache.
+    - ``8898``
+  * - ``hive.cache.bookkeeper-port``
+    -  The TCP/IP port used by the BookKeeper managing the cache.
+    - ``8899``
+  * - ``hive.cache.parallel-warmup-enabled``
```
Comment (sopel39) [inline presto-docs/src/main/sphinx/connector/hive-storage-caching.rst:112]: that is called `hive.cache.read-mode` now and is enum of values:
```
async
read-through
```

File: presto-docs/src/main/sphinx/connector/hive-storage-caching.rst (hunk 8 above)
Comment (mosabua) [reply presto-docs/src/main/sphinx/connector/hive-storage-caching.rst:63]: Updated in new PR

**Pull request closed without merging**